### ISSUE TITLE
Timeout 120 seconds for step RHYTHMS

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#104]
![and so it goes_03](https://user-images.githubusercontent.com/33522630/36945977-08eea7e2-1fb6-11e8-914b-9810096ea74b.jpg)

The appended image leads to a time-out in step RHYTHMS.
Unclear, why. The transcription seems to be rather good, nothing in rose.

**Comment by maximumspatium** [comment]
> Unclear, why. The transcription seems to be rather good, nothing in rose.

An "unsolvable" rhythmical pattern has been encountered. Such cases don't happen often but if it happens it usually leads to an endless loop in the RHYTHM step.

**Comment by hbitteur** [comment]
This is the reason why there is a default time-out on each step of the engine.
Now, I will investigate on this precise example why the rhythms step has not detected an endless loop.

**Comment by hbitteur** [comment]
I found several problems.

Time-out was due to voice assignment, fixed by commit 30bff6a96eb437c014ebe6199b60e625b8637f67 which now limits voice computation to measure (part level) even though time slot assignment is still performed within measure stack (system level).

Another problem was due to false positive in detection of small heads (cue notes), leading to the removal of a real barline, which however got later "rebuilt" thanks to the voting mechanism within columns of system barlines. Unfortunately the "rebuilding" had a bug, which is now fixed in commit 7da2a9173a0d3e4a32bfa43ed8ecd53fdb5bffbb

Note that if the images you transcribe have no cue notes, it is safer and more efficient to disable cue notes detection in OMR engine.
This can be done today by setting the constant `ShapeSet.supportSmallHeadNotes` to false (the application must be restarted for this global modification to be applied).
I'm thinking of enabling the user to interactively enable/disable some of these features at book or even sheet level, this would be a more convenient way.
